### ISSUE TITLE
02 - Validation Accuracy ≫ Test Accuracy

### ISSUE THREAD

**Issue by anacarsi** [anacarsi/ss25_Hierarchical_Multiscale_Image_Classification#2]
Different experiments were run to determine the accuracy of the model. For the next experiments, we skip evaluating on Test Accuracy exclusively, since it is not reliable for this imbalanced dataset.

Current reasoning on why this occurs:

- The validation set does not reflect the true class  distribution. The validation set was balanced to reduce the risk of the model guessing always "no tumor" and achieving a high accuracy.
- Experiments were done as well comparing training with **weighted loss** Vs. a balanced training set. Balanced Training + Balanced Validation = Overestimation, because this setup inflates the model's confidence. 
- Insufficient regularization may be a problem for achieving Training Acc. 1.00 in some cases. I assume enough diversity due to plenty of transformations and > 200.000 real patches.
- The model **spams** positives (Test Precision 0.43) -> because it learned in 50/50 situations?

Experiments:
E1	Balanced	Natural (unbalanced)	CE	Common setup
E2	Unbalanced	Natural	Weighted CE	Better simulates deployment
E3	Balanced	Balanced	CE	Inflated metrics, for contrast only
E4	Unbalanced	Balanced	Weighted CE	Hybrid approach
E5	Unbalanced + Aug	Natural	Focal Loss	Improves minority class focus
E6    **Mean- Max- Attention- Pooling** Comparison
